glean-core/tests/counter.rs

```diff
@@ -0,0 +1,152 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+mod common;
+use crate::common::*;
+
+use glean_core::metrics::*;
+use glean_core::storage::StorageManager;
+use glean_core::{CommonMetricData, Glean, Lifetime};
+
+// SKIPPED from glean-ac: counter deserializer should correctly parse integers
+// This test doesn't really apply to rkv
+
+#[test]
+fn counter_serializer_should_correctly_serialize_counters() {
+    let (_t, tmpname) = tempdir();
+    {
+        let glean = Glean::new(&tmpname, GLOBAL_APPLICATION_ID).unwrap();
+
+        let metric = CounterMetric::new(CommonMetricData {
+            name: "counter_metric".into(),
+            category: "telemetry".into(),
+            send_in_pings: vec!["store1".into()],
+            disabled: false,
+            lifetime: Lifetime::User,
+            ..Default::default()
+        });
+
+        metric.add(&glean, 1);
+
+        let snapshot = StorageManager
+            .snapshot_as_json(glean.storage(), "store1".into(), true)
+            .unwrap();
+        assert_eq!(
+            "{\"counter\":{\"telemetry.counter_metric\":1}}",
```

Comment:
Instead of constructing a string we should be able to build the JSON value like:

```
json!({"counter": {"telemetry.counter_metric": 1}})
```

and compare it to  `snapshot`.

Reply:
In this case I actually think it's valuable to test against a string -- we want to make sure we're serializing with no unnecessary whitespace and in a single line etc.  Could move that out into its own test instead, of course...

Reply:
But the string-serialization happens in this test, not in library code. We therefore test only that `serde_json::to_string` kinda does what we expect, but don't test that anything internally actually uses that.

Reply:
Ah, good point. :slightly_frowning_face: Indeed, makes more sense to compare JSON objects then.